File: DynamicProgramming/357-CountNumberswithUniqueDigits.py
class Solution(object):

    def countNumbersWithUniqueDigits(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n >= 11:
            return 0
        if n == 0:
            return 1
        sumb, f = 10, 9
        for i in range(2, n + 1):
            f = f * (11 - i)
            sumb = sumb + f
        return sumb

File: DynamicProgramming/357-CountNumberswithUniqueDigits.java
public class Solution {
    public int countNumbersWithUniqueDigits(int n) {
        if(n < 0 || n >= 11) return 0;
        if(0 == n) return 1;
        int sum = 10;
        int f = 9;
        for(int i=2; i<=n; i++) {
            f *= (11-i);
            sum += f;
        }
        return sum;
    }
}